docs/antrea-network-policy.md

```diff
@@ -659,6 +659,9 @@ for all NetworkPolicies in the Namespace. Packets of any connection that match
 a NetworkPolicy rule will be logged with a reference to the NetworkPolicy name,
 but packets dropped by the implicit "default drop" (not allowed by any NetworkPolicy)
 will only be logged with consistent name `K8sNetworkPolicy` for reference.
+Note that currently, Antrea only retrieves the logging Annotation once when adding
+NetworkPolicies and in case of agent restart, users should not update Namespace
+logging Annotations, otherwise it would risk NetworkPolicies working in a stale state.
```

Comment:
As a reader (a probably bad one :)) I understand that if an agent is restarted, users should not be updated the namespace logging annotation anymore.

If I am not mistaken, the concept to convey is that events where the namespace annotation is processed concurrently with the network policy, such as agent restart, might trigger a risk of having stale NetworkPolicy info realized. If my understanding is correct, we might consider a little rephrasing above.

Reply:
@salv-orlando The race condition was explained here: https://github.com/antrea-io/antrea/pull/4047#discussion_r937933519. I think we should remove this restriction in this release. It would be a simple change after https://github.com/antrea-io/antrea/pull/4028 is merged. I was waiting for these feature PRs to be merged first, otherwise all of them need to address many conflicts caused by the refactoring by #4028

Reply:
I do not get what exact issues can happen under what exact conditions either. Could you explain more? @qiyueyao

Reply:
I guess we are talking about two issues here:

1. We do not handle annotation changes after a NetworkPolicies are created.
2. For agent restart (before? after? during?), users cannot change the Namespace annotation, otherwise we can run into some issues (but what issues? What does "working in a stale state means"?

Reply:
Discussed offline and opened follow up PR #4099.